de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/user/api/IUserDAO.java

```diff
@@ -69,11 +69,13 @@
 
 	/**
 	 * Fetch all system(login) user IDs
-	 * 
+	 *
 	 * @param ctx
 	 * @return AD_User_IDs
 	 */
 	List<Integer> retrieveSystemUserIds();
 
 	BPartnerId getBPartnerIdByUserId(final UserId userId);
+
+	<T extends org.compiere.model.I_AD_User> T getByIdInTrx(int userId, Class<T> modelClass);
```

Comment:
when adding new methods like this one, pls don't use `int` parameters, but the respective `RepoIdAware`.
Sidenote: if it doesn't yet exist, please consider creating it

Reply:
Done